terraform/monitoring/03_service_slo.tf

```diff
@@ -0,0 +1,100 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Here we create a custom service that we attach our SLOs to
+# Using a custom service here allows us to add on additional SLOs and 
+# alerting policies on things such as custom metrics
+resource "google_monitoring_custom_service" "frontend" {
+  service_id = "frontend-srv"
+  display_name = "Frontend Service"
+}
+
+# Here we define an SLO for availablity for the frontend service 
+# The SLO is defined as following: 
+#   80% of all non-4XX requests within the past 30 day windowed period
+#   return with 200 OK status
+resource "google_monitoring_slo" "frontend_availability_slo" {
+  service = google_monitoring_custom_service.frontend.service_id
+  slo_id = "availability-slo"
+  display_name = "Availability SLO with request base SLI (good total ratio)"
+
+  # The goal sets our 80% objective with a 30 day rolling window period
```

Comment:
Why 80%?
If we think the error budget will be burned too quickly, we should at least add a comment to explain that this is not the best practice but we do it for educational purposes.

Reply:
Agreed.  I mentioned this to Nicholas yesterday.  The exact numbers don't matter here too much since it's for educational purposes.

Reply:
Yup :) added a comment explaining the reason.